src/callbacks_step/save_restart_dg.jl

```diff
@@ -327,4 +327,28 @@ function load_restart_file_on_root(mesh::Union{ParallelTreeMesh, ParallelP4estMe
 
     return u_ode
 end
+
+function save_restart_controller(integrator,
+                                 controller, restart_callback)
+    # Save only on root
+    if mpi_isroot()
+        @unpack output_directory = restart_callback
+        timestep = integrator.stats.naccept
+
+        # Filename based on current time step
+        filename = joinpath(output_directory, @sprintf("restart_%06d.h5", timestep))
+
+        # Open file (preserve existing content)
+        h5open(filename, "cw") do file
+            # Add context information as attributes
+            attributes(file)["qold"] = integrator.qold
+            # Ensure that `dtpropose` is written as a double precision scalar
+            attributes(file)["dtpropose"] = convert(Float64, integrator.dtpropose)
+            # For PIDController is necessary to save additional parameters
+            if :err in fieldnames(typeof(controller))
+                attributes(file)["controller_err"] = controller.err
+            end
```

Comment:
Maybe a simpler way to implement this would be with [`hasproperty`](https://docs.julialang.org/en/v1/base/base/#Base.hasproperty)?
```suggestion
            if hasproperty(controller, :err)
                attributes(file)["controller_err"] = controller.err
            end
```